src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -140,6 +140,8 @@ internal override FlowAnalysisAnnotations FlowAnalysisAnnotations
                 return DecodeFlowAnalysisAttributes(GetDecodedWellKnownAttributeData());
             }
         }
+        internal override bool IsNullChecked
+            => this.CSharpSyntaxNode.ExclamationToken.Kind() == SyntaxKind.ExclamationToken;
```

Comment:
Can ExclamationToken be `null`? Also, I think the standard way of checking if a token is present is to call `IsMissing` instead of checking the `Kind`.

Reply:
I believe it's SyntaxKind.None if there's no token.

Reply:
I'm also wondering about when `IsMissing` is preferred over `Kind() == SyntaxKind.None`. It seems like IsMissing is mainly for when the token's absence is a syntax error, not when the token is optional.